File: src/main/com/jatti/entity/EntityCreate.java
package com.jatti.entity;

import org.bukkit.Bukkit;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntitySpawnEvent;

public class EntityCreate implements Listener{

    @EventHandler
    public void onSpawn(EntitySpawnEvent evt){

        Entity.Companion.get(EntitiesUtils.getEntities().size()+1).onCreate(evt.getLocation());

    }

    @EventHandler
    public void onCreate(MEntityCreateEvent evt){

        //TODO Add changing org.bukkit.entity.EntityType for Entities
        //TODO Add special abilities for Entities
        //TODO Add mini bosses

    }
}

File: src/main/com/jatti/entity/EntityCreate.kt
package com.jatti.entity

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntitySpawnEvent

class EntityCreate : Listener {

    @EventHandler
    fun onSpawn(evt: EntitySpawnEvent) {

        if (evt.entityType == EntityType.ZOMBIE.type) {

            val e: Entity = Entity.get(EntitiesUtils.getEntities().size + 1)

            evt.entity.remove()
            e.onCreate(evt.location)
        }

    }

    @EventHandler
    fun onCreate(evt: MEntityCreateEvent) {

        evt.entity.shuffle()
        println("Nowy entity na: " + evt.location + " Typ: " + evt.entity.type)
        //TODO Add special abilities for Entities
        //TODO Add mini bosses
    }
}